docs/basics/101-127-procedures.rst

```diff
@@ -0,0 +1,361 @@
+:command:`datalad create` as
+
+- ``datalad create -c yoda <DSname>``
+- ``datalad create -c text2git <DSname>``
+- ``datalad create -c metadatatypes <DSname>``
+
+As a general note, it is useful to apply procedures
+as early as possible. Procedures such
+as ``cfg_yoda`` (explained in detail in section :ref:`yoda`),
+assume clean, freshly created datasets when they run,
+and given that many procedures create files or change ``.gitattributes``,
+existing files or configurations in a dataset can lead to unexpected
+problems or failures.
+
+
+.. findoutmore:: Write your own procedures
+
+   Procedures can come with source code or datasets, but anyone can
+   write their own extension as well, if they wish. This allows to
+   automate routine configurations or tasks in a dataset.
+   Some general rules for creating a custom procedure are outlined
+   below:
+
+     .. todo::
+
+        turn shebang into term once PR#134 is merged
+
+   - A procedure can be any executable. Executables must have the
+     appropriate permissions and, in the case of a script,
+     must contain an appropriate shebang
+
+       - If a procedure is not executable, but its filename ends with
+         ``.py``, it is automatically executed by the Python interpreter
+         (whichever version is available in the present environment).
+         Likewise, procedure implementations ending on ``.sh`` are executed
+         via :term:`bash`.
+
+   - Procedures can implement any argument handling, but must be capable
+     of taking at least one positional argument (the absolute path to the
+     dataset they shall operate on).
+
+   - Custom procedures rely heavily on configurations in ``.datalad/config``
+     (or the associated environment variables). Within ``.datalad/config``,
+     each procedure should get an individual entry that contains at least
+     a short "help" description on what the procedure does. Below is a minimal
+     ``.datalad/config`` entry for a custom procedure:
+
+     .. code-block:: bash
+
+        [datalad "procedures.<NAME>"]
+           help = "This is a string to describe what the procedure does"
+
+   - By default, DataLad will search for user procedures (i.e. procedures on the
+     *global* level) in ``~/.config/datalad/procedures``, and for dataset procedures
+     (i.e. the *local* level) in ``.datalad/procedures`` relative to a dataset root.
+     Note that ``.datalad/procedures`` does not exist by default, and the ``procedures``
+     directory needs to be created first.
+     Alternatively to the default locations, DataLad can be pointed to the
+     location of a procedure with a configuration in ``.datalad/config``,
+     or with the help of :term:`environment variable`\s.
+     The appropriate configuration keys for
+     ``.datalad/config`` are either ``datalad.locations.system-procedures``
+     for changing the *global* default or ``datalad.locations.dataset-procedures`` to
```

Comment:
This is somewhat messing up the terms *global* and *local* with the way they are used by `git-config`. The latter talks about *system*, *global* and *local*. We introduce *dataset* in addition. Plus there's `datalad.locations.user-procedures` missing (and that is what would be "global"). "local" in the sense of repo specific but not committed doesn't exist as far as procedure locations go.